src/main/java/AddCommand.java

```diff
@@ -0,0 +1,72 @@
+package duke.command;
+
+import duke.error.DukeException;
+import duke.task.Task;
+import duke.task.Deadline;
+import duke.task.Todo;
+import duke.task.Event;
+import duke.Parser;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+
+/**
+ * Deals with operation to add new tasks.
+ */
+public class AddCommand extends Command {
+    protected String command;
+    protected Task task;
+    protected String type;
+
+    /**
+     * Initializes AddCommand object.
+     * Checks if command is valid before assignment object.
+     *
+     * @param type Type of task
+     * @param command String user input of the task details
+     * @throws DukeException if command is invalid
+     */
+    public AddCommand(String type, String command) throws DukeException {
+        try {
+            Parser.checkTask(type, command);
+            if (type.equals("T")) {
+                this.task = new Todo(command);
+                this.type = "T";
+            } else if (type.equals("D")) {
+                String[] arr = command.split("/by");
+                this.task = new Deadline(arr[0].trim(), Parser.datetimeformat(arr[1].trim()));
+                this.type = "D";
+            } else if (type.equals("E")) {
+                String[] arr = command.split("/at");
+                this.task = new Event(arr[0].trim(), arr[1].trim());
+                this.type = "E";
+            } else {
+                throw new AssertionError("ERROR: Invalid Task Type!");
+            }
+        } catch (DukeException e) {
```

Comment:
Perhaps you don't need to `catch` the `DukeException` to throw it again? You can just `throw` the exception directly.